gcloud/orb.yaml

```diff
@@ -483,5 +483,56 @@ jobs:
           image: <<parameters.project>>/<<parameters.image>>
           registry: <<parameters.registry>>
 
+  artifacts-package-publish:
+    description: >
+      This job will publish an already built package to a GCP package
+      repository.
+    docker:
+      - image: <<parameters.executor>>
+    resource_class: <<parameters.resource_class>>
+    parameters:
+      creds:
+        default: GCLOUD_SERVICE_KEY
+        description: >
+          Name of environment variable storing the base64-encoded service key
+          for the GCP project.
+        type: env_var_name
+      executor:
+        default: python:3.7.10
+        description: >
+          Name of the python image to use to execute the job.
+        type: string
+      path:
+        default: .
+        description: >
+          Path to the directory containing your packages.
+        type: string
+      project:
+        description: >
+          Name of GCP project to which we will push.
+        type: string
+      resource_class:
+        default: medium
+        type: string
+      repository-url:
+        description: >
+          URL of GCP package repository to which we will push.
+        type: string
+      workspace:
+        default: '/tmp/poetry_build'
+        description: >
+          If specified, attaches the specified workspace and copies it into the
+          image context before building.
+        type: string
+    steps:
+      - run: pip install twine keyring keyrings.google-artifactregistry-auth
```

Comment:
I'm surprised we need to use `twine` for this upload, given the `workspace` here implies poetry compatibility. Is there a blocker to solving this via the `poetry/publish` job ([source](https://github.com/talkiq/circleci-orbs/blob/master/poetry/orb.yaml#L37))?

My hope is that we could directly `poetry publish -r $GCP_AR_URL` with all the same commands there; maybe by adding eg.
```yaml
- publish:
    # SNIP
    pre-steps:
      - run: python3 -m pip install keyring keyrings.google-artifactregistry-auth
      - gcloud/install
      - gcloud/auth:
          # SNIP
```
etc.

Reply:
I'm just waiting for the PR https://github.com/python-poetry/poetry/pull/4086 to get merged. In the meantime, I though I could use `twine` to start trying to upload content to test GCP's reliability.

When the poetry PR is merged, we can actually just do `poetry publish -r $GCP_AR_URL` as you suggest. I just don't know how long that will take.

Reply:
Ahhh, gotcha! Alright, I'm thinking let's just keep this inline in the repo we're testing out, then -- no need to release this as an OSS orb if it's just temporary for testing. Note that orbs can be entirely inline in any CI config file (or this could just directly be a job there).

Reply:
Sounds good. I'll take that approach.

Reply:
There is a production and a staging repository in GCP. And while I could just do everything ad hoc, it really feels like having this functionality in the orb might be valuable if we ever add other repositories.

We don't need to use `twine` forever, just until `poetry` finalizes the PR.

Reply:
Oh, I might be thinking of things incorrectly here -- is the plan to actually rely on this for a variety of services until poetry is ready, or is the intention more to test with a single repo to verify functionality? If it's the latter, I think keeping this consolidated to the given repo will be easier to work with as we won't need to coordinate across multiple repos or worry about OSS-level extensibility, but if it's more the former then I agree having it here is correct.

Will review assuming its the former, but please let me know your thoughts!

Reply:
Okay, I'm having second thoughts about this. If this is going to be anything, it really should be a command. I'll explore that route because that feels a bit more straightforward than mounting and unmounting workspaces.

Reply:
@Kevin,

I have converted the job into a command.